pkg/cookies/cookies.go

```diff
@@ -30,12 +30,15 @@ func MakeCookieFromOptions(req *http.Request, name string, value string, opts *o
 		Value:    value,
 		Path:     opts.Path,
 		Domain:   domain,
-		Expires:  now.Add(expiration),
 		HttpOnly: opts.HTTPOnly,
 		Secure:   opts.Secure,
 		SameSite: ParseSameSite(opts.SameSite),
 	}
 
+	if expiration != time.Duration(0) {
+		c.Expires = now.Add(expiration)
+	}
+
```

Comment:
How does this change affect the session refreshing? Have you checked that area of the code to see if there's any interactions there?

Reply:
Yep.
I checked and tested that theres no change in session refreshing mechanism.

This is the function which judges if session should be refreshed or not.

https://github.com/oauth2-proxy/oauth2-proxy/blob/f753ec1ca5c42cd4c4e400afa1f76817638f3010/pkg/middleware/stored_session.go#L200-L203

We have to think about the value of `session.Age()` for session refreshing.
As for cookies, this value have nothing to do with the value of cookies' Expire.

Cookie's age is calculated by `current time - cookie created time`.
Here, created time is the timestamp saved within the cookie's value at the time of  cration (= `timeStr` below).

https://github.com/oauth2-proxy/oauth2-proxy/blob/f753ec1ca5c42cd4c4e400afa1f76817638f3010/pkg/encryption/utils.go#L82

This is the overview of the session refreshing mechanism.
So, I believe this change does not affect on the session refreshing mechanism.